platform/onebox/backup_and_restore/ceph/get-credentials.sh

```diff
@@ -5,11 +5,24 @@ This scipt will generate json file for creds to be used with import export
 Script will generate file storage-creds.json
 Run it from the VM running aifabric.
 Use it as it is [insecure] or transfer to some credsManager and then change backup/restore scripts to fetch from credsmanager instead of json file
+# $1 - [Optional but recommended] pass private ip of the aif machine on which it is accesible from other vms in the same network
 [Script Version -> 21.4]
 '
+
+readonly PRIVATE_IP=$1
+
 function initialize_variables() {
-	# Gets private ip of machine so that it can be connected within the VM
-	OBJECT_GATEWAY_EXTERNAL_HOST=$(hostname -i)
+	if [ -z "$PRIVATE_IP" ]; then
+		# Gets private ip of machine so that it can be connected within the VM
+		# Seems to be set as localhost on some customer machines
+		#OBJECT_GATEWAY_EXTERNAL_HOST=$(hostname -i)
+		PRIVATE_ADDRESS=$(ip -o route get to 8.8.8.8 | sed -n 's/.*src \([0-9.]\+\).*/\1/p')
+    	#This is needed on k8s 1.18.x as $PRIVATE_ADDRESS is found to have a newline
+    	OBJECT_GATEWAY_EXTERNAL_HOST=$(echo "$PRIVATE_ADDRESS" | tr -d '\n')
+    else
+    	OBJECT_GATEWAY_EXTERNAL_HOST=$PRIVATE_IP
```

Comment:
nit: lets put an echo statement here saying Host is set to this

Reply:
makes sense, will do later, basically echo what user passed and what was set